openedx_events/event_bus/avro/schema.py

```diff
@@ -0,0 +1,95 @@
+"""
+Code to convert attr classes to Avro specification.
+
+TODO (EventBus): handle optional parameters and allow for schema evolution (ARCHBOM-2013)
+"""
+from datetime import datetime
+
+from opaque_keys.edx.keys import CourseKey
+
+from .custom_serializers import CourseKeyAvroSerializer, DatetimeAvroSerializer
+from .types import PYTHON_TYPE_TO_AVRO_MAPPING
+
+DEFAULT_FIELD_TYPES = {
+    datetime: DatetimeAvroSerializer.field_type,
+    CourseKey: CourseKeyAvroSerializer.field_type,
+}
+
+
+def schema_from_signal(signal, custom_field_types=None):
+    """Create an Avro schema for events sent by an instance of OpenEdxPublicSignal."""
```

Comment:
Should we add an assertion or type annotation that the signal is expected to be an OpenEdxPublicSignal?

Reply:
Resolved with parameter docs